From fe93feef4ec94926b0ccefb3ab4973ef9331b22b Mon Sep 17 00:00:00 2001
From: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Date: Wed, 23 Apr 2014 14:53:45 +0100
Subject: [PATCH 198/351] switch: arizona: Move callback out of
 arizona_hpdet_read

arizona_hpdet_read should only contain the handling for reading the
impedance value move the handling for the callback and storing the
impedance value out into the IRQ handler.

Change-Id: I94a94b1d0597ac77fbc6366420ccf146d79573a2
Signed-off-by: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 drivers/switch/switch-arizona.c |   11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/drivers/switch/switch-arizona.c b/drivers/switch/switch-arizona.c
index 0238aaa..6665532 100644
--- a/drivers/switch/switch-arizona.c
+++ b/drivers/switch/switch-arizona.c
@@ -590,12 +590,8 @@ static int arizona_hpdet_read(struct arizona_extcon_info *info)
 		}
 	}
 
-	arizona->hp_impedance = val;
-
-	if (arizona->pdata.hpdet_cb)
-		arizona->pdata.hpdet_cb(arizona->hp_impedance);
-
 	dev_dbg(arizona->dev, "HP impedance %d ohms\n", val);
+
 	return val;
 }
 
@@ -820,6 +816,11 @@ static irqreturn_t arizona_hpdet_irq(int irq, void *data)
 	}
 	reading = ret;
 
+	arizona->hp_impedance = reading;
+
+	if (arizona->pdata.hpdet_cb)
+		arizona->pdata.hpdet_cb(arizona->hp_impedance);
+
 	/* Reset back to starting range */
 	regmap_update_bits(arizona->regmap,
 			   ARIZONA_HEADPHONE_DETECT_1,
-- 
1.7.9.5

